src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.cs

```diff
@@ -57,13 +57,43 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
                 context.SyntaxProvider
 
                 // Find all MethodDeclarationSyntax nodes attributed with GeneratedRegex and gather the required information.
+                // The predicate will be run once for every attributed node in the same file that's being modified.
+                // The transform will be run once for every attributed node in the compilation.
```

Comment:
> // The transform will be run once for every attributed node in the compilation.

This comment is not correct. Atleast I really really really hope so ;)
In the appropriate comment in https://github.com/dotnet/roslyn/blob/e37327dd23741dae3d35338b8d7020442c1fd1ce/src/Compilers/Core/Portable/SourceGeneration/Nodes/SyntaxValueProvider_ForAttributeWithMetadataName.cs#L74-L76 it is stated that_
> A function that performs the transform. This will only be passed nodes that return 'true' for predicate and which have a  matching 'AttributeData' whose 'AttributeData.AttributeClass' has the same fully qualified, metadata name as 'fullyQualifiedMetadataName'

(I removed a few of the refs in the comment to make it readable).

This should be changed to "The transform will be run once for every attributed node returned by the predicate" or similar.

Reply:
By "attributed node" I meant with the generated regex attribute. I believe that's a correct statement. What's wrong with it?

Reply:
It's not the compilation, the compilation is a lot more than just the changed files. The predicate runs on the changed files as your comment for ``predicate`` states. The transformation then runs on the data returned by the predicate. And that set of data is lot less then the full compilation.

Reply:
tl;dr: Your comment is correct, sorry for the noise.
I've reread (and reread again) the document about incremental source gens.

> It is because of this split that performance can be achieved: as the driver is aware of which nodes are chosen for examination, it can safely skip the first predicate lambda when a syntax tree remains unchanged. The driver will still re-run the second transform lambda even for nodes in unchanged files, as a change in one file can impact the semantic meaning of a node in another file.

https://github.com/dotnet/roslyn/blob/main/docs/features/incremental-generators.md